### ISSUE TITLE
✅Qiqqa lagging when terminating: FolderWatcher takes forever on large Watch dirtree

### ISSUE THREAD

**Issue by GerHobbelt** [jimmejardine/qiqqa-open-source#50]
HTR:

- have a directory tree with many files and directories, which is *watched* by Qiqqa for new PDFs to occur.
- Quit Qiqqa

Logging will report something like this:

```
20190814.130928 [Q] DEBUG [Daemon.Maintainable:GeneralTaskDaemon.DoMaintenance] [187.176M] 
      Directory.GetFiles took 300433 ms
```

(augmented Qiqqa @ commit XXX)

Note the **300 seconds** it took GetFiles() to complete.

**Comment by GerHobbelt** [comment]
Fixed in commit SHA-1: 08af03eaf53b56853beed0e35b73987545adc435

using `EnumerateFiles()` API instead of `GetFiles()`. 

TODO: inspect other sites in code where `GetFiles` is invoked.

**Comment by GerHobbelt** [comment]
> Closing <sup>and decluttering the issue list so it stays workable for me</sup>: fixed in https://github.com/GerHobbelt/qiqqa-open-source mainline=master branch, pending #15 / any maintainer rights/actions.
